src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -404,82 +455,82 @@ protected override void PostStop()
-                 * listen.AddressPromise.
-                 * */
-                .With<Listen>(listen => Listens.ContinueWith<INoSerializationVerificationNeeded>(listens =>
+            /*
+            * the first command the EndpointManager receives.
+            * instructs the EndpointManager to fire off its "Listens" command, which starts
+            * up all inbound transports and binds them to specific addresses via configuration.
+            * those results will then be piped back to Remoting, who waits for the results of
+            * listen.AddressPromise.
+            * */
+            Receive<Listen>(listen => Listens.ContinueWith<INoSerializationVerificationNeeded>(listens =>
+            {
+                if (listens.IsFaulted)
                 {
-                    if (listens.IsFaulted)
-                    {
-                        return new ListensFailure(listen.AddressesPromise, listens.Exception);
-                    }
-                    else
-                    {
-                        return new ListensResult(listen.AddressesPromise, listens.Result);
-                    }
-                }, TaskContinuationOptions.ExecuteSynchronously)
-                    .PipeTo(Self))
-                .With<ListensResult>(listens =>
+                    return new ListensFailure(listen.AddressesPromise, listens.Exception);
+                }
+                else
                 {
-                    _transportMapping = (from mapping in listens.Results
-                                        group mapping by mapping.Item1.Address
-                                            into g
-                                            select new { address = g.Key, transports = g.ToList() }).Select(x =>
-                        {
-                            if (x.transports.Count > 1)
-                            {
-                                throw new RemoteTransportException(
-                                    string.Format("There are more than one transports listening on local address {0}",
-                                        x.address));
-                            }
-                            return new KeyValuePair<Address, AkkaProtocolTransport>(x.address,
-                                x.transports.Head().Item1.ProtocolTransport);
-                        }).ToDictionary(x => x.Key, v => v.Value);
-
-                    //Register a listener to each transport and collect mapping to addresses
-                    var transportsAndAddresses = listens.Results.Select(x =>
-                    {
-                        x.Item2.SetResult(new ActorAssociationEventListener(Self));
-                        return x.Item1;
-                    }).ToList();
+                    return new ListensResult(listen.AddressesPromise, listens.Result);
+                }
+            }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default)
```

Comment:
@Aaronontheweb @JeffCyr just so I follow this, doesnt `TaskScheduler.Default` mean that the continuation will be scheduled on the default thread pool?
and how does that work together with `ExecuteSynchronously` ?
If `TaskScheduler.Default` means the default thread pool, we might end up running remoting actors together with noisy user actors, which was the reason why we moved away from the thread pool in the first place.

https://msdn.microsoft.com/en-us/library/dd997402(v=vs.110).aspx
>The default scheduler for Task Parallel Library and PLINQ uses the .NET Framework ThreadPool to queue and execute work. In the .NET Framework 4, the ThreadPool uses the information that is provided by the System.Threading.Tasks.Task type to efficiently support the fine-grained parallelism (short-lived units of work) that parallel tasks and queries often represent.

Am I wrong here?